examples/react/large-file-based/src/routeTree.gen.ts

```diff
@@ -127,18 +127,97 @@ declare module '@tanstack/react-router' {
 
 // Create and export the route tree
 
-export const routeTree = rootRoute.addChildren({
-  IndexRoute,
-  ParamsRouteRoute: ParamsRouteRoute.addChildren({
-    ParamsParamsPlaceholderRoute,
-  }),
-  SearchRouteRoute: SearchRouteRoute.addChildren({
-    SearchSearchPlaceholderRoute,
-  }),
-  AbsoluteRoute,
-  LinkPropsRoute,
-  RelativeRoute,
-})
+const IndexRouteWithChildren = IndexRoute
+const ParamsParamsPlaceholderRouteWithChildren = ParamsParamsPlaceholderRoute
+interface ParamsRouteRouteChildren {
```

Comment:
We now declare children as an interface. This means basically just delays the inference until you index into the interface